test/int/stackTrace.test.ts

```diff
@@ -55,37 +58,50 @@ function assertSourceMatches(actual: DebugProtocol.Source | undefined, expected:
         url.pathname = expected.urlRelativePath;
         expectedName = url.host;
         expectedPath = url.toString();
+    } else if (expected.evalCode === true) {
+        // Eval code has source that looks like 'VM123'. Check it by regex instead.
+        expect(actual.name).to.match(/.*VM.*/, `Frame ${index} source name`);
+        expect(actual.path).to.match(/.*VM.*/, `Frame ${index} source path`);
+        return;
     } else {
-        assert.fail('Not enough information for expected source: set either "fileRelativePath" or "urlRelativePath"');
+        assert.fail('Not enough information for expected source: set either "fileRelativePath" or "urlRelativePath" or "eval"');
         return;
     }
 
-    assert.equal(actual.name, expectedName, `Source name for frame ${index} does not match`);
-    assert.equal(actual.path, expectedPath, `Source path for frame ${index} does not match`);
+    expect(actual.name).to.equal(expectedName, `Frame ${index} source name`);
+    expect(actual.path).to.equal(expectedPath, `Frame ${index} source path`);
 }
 
 function assertFrameMatches(actual: DebugProtocol.StackFrame, expected: ExpectedFrame, index: number) {
-    assert.equal(actual.name, expected.name, `Name for frame ${index} does not match`);
-    assert.equal(actual.line, expected.line, `Line number for frame ${index} does not match`);
-    assert.equal(actual.column, expected.column, `Column number for frame ${index} does not match`);
+    if (expected.name) {

```

Comment:
What do you think of using name for both string and regexp, and using typeof or instance of to figure out which assertion to use?

Reply:
I think this is a good idea in this case. I fixed this.